### PR TITLE
feat(search): Update docs to reflect new my_teams filter

### PR DESCRIPTION
**Pull request by 17hogeju**
Updates the documentation for the changes made for https://github.com/getsentry/sentry/issues/50037

- New issue filter tag `my_teams` only references teams user belongs to
- Previous tag `me` now only refers to issues directly assigned to user
- To recreate the older functionality of `me`, user can type `[me, my_teams]`

Internal update: 06/26/2023
Target GA: 07/05/2023

### CODE REVIEW

File: src/docs/product/sentry-basics/search/searchable-properties.mdx [hunk 1]
```diff
@@ -89,8 +89,8 @@ Release properties overlap with event and issue search properties, so they don't
...
+| assigned | Returns issues assigned to the defined user(s) or team(s). Values can be a user ID (your email address), `me` for yourself, `me_or_none` for yourself or no assignee, `my_teams` for teams you belong to, or # followed by the team name, that is `#team-name`. |  | x |  | team or org user |
+| assigned_<br></br>or_suggested | Returns issues assigned to or suggested to be assigned to the defined user(s) or team(s). Suggested assignees are determined by matching [ownership rules](/product/issues/ownership-rules/) and [suspect commits](/product/issues/suspect-commits/). Values can be a user ID (your email address), `me` for yourself, `me_or_none` for yourself or no assignee/suggestions, `my_teams` for teams you belong to, or # followed by the team name, that is `#team-name`. |  | x |  | team or org user |
```
Comment (lizokm) [inline src/docs/product/sentry-basics/search/searchable-properties.mdx:93]:
```suggestion
| assigned_<br></br>or_suggested | Returns issues that are either assigned to or suggested as something that should be assigned to defined user(s) or team(s). Suggested assignees are found by matching [ownership rules](/product/issues/ownership-rules/) with [suspect commits](/product/issues/suspect-commits/). Values can be a user ID (your email address), `me` for yourself, `me_or_none` for yourself or no assignee/suggestion, `my_teams` or `#team-name` for teams you belong to. |  | x |  | team or org user |
```

**Review by lizokm — APPROVED** [review]
Just a few small wording tweaks.

**Comment by shanamatthews** [comment]
Hey @17hogeju - is this ready to merge?

**Comment by 17hogeju** [comment]
> Hey @17hogeju - is this ready to merge?

We will merge this shortly after GA

File: src/docs/product/sentry-basics/search/searchable-properties/issues.mdx [hunk 2]
```diff
@@ -26,13 +26,13 @@ Indicates if the app is in the foreground or background. Values are `1/0` or `tr
 
 ### `assigned`
 
-Returns issues assigned to the defined user. Values can be a user ID (your email address), `me` for yourself, `me_or_none` for yourself or no assignee, or # followed by the team name, that is `#team-name`.
+Returns issues assigned to the defined user(s) or team(s). Values can be a user ID (your email address), `me` for yourself, `me_or_none` for yourself or no assignee, `my_teams` or `#team-name` for teams you belong to.
```
Comment (barkbarkimashark) [inline src/docs/product/sentry-basics/search/searchable-properties/issues.mdx:29]: Hmm, `me_or_none` might be confusing for the user to use. I'm not sure thats a legal value a user can enter. 

Would something like:
```
Values can be a user ID (your email address), `me` for yourself, `my_teams` for teams you belong to, `#team-name` for a specific team assigned to issues, or `none` for issues that aren't assigned.
```

File: src/docs/product/sentry-basics/search/searchable-properties/issues.mdx [hunk 3]
```diff
@@ -26,13 +26,13 @@ Indicates if the app is in the foreground or background. Values are `1/0` or `tr
...
+Returns issues assigned to the defined user(s) or team(s). Values can be a user ID (your email address), `me` for yourself, `none` for no assignee, `my_teams` or `#team-name` for teams you belong to.
 
 - **Type:** team or org user
 
 ### `assigned_or_suggested`
 
-Returns issues assigned to or suggested to be assigned to the defined user. Suggested assignees are determined by matching [ownership rules](/product/issues/ownership-rules/) and [suspect commits](/product/issues/suspect-commits/). Values can be a user ID (your email address), `me` for yourself, `me_or_none` for yourself or no assignee/suggestions, or # followed by the team name, that is `#team-name`.
+Returns issues that are either assigned to or suggested as something that should be assigned to defined user(s) or team(s). Suggested assignees are found by matching [ownership rules](/product/issues/ownership-rules/) with [suspect commits](/product/issues/suspect-commits/). Values can be a user ID (your email address), `me` for yourself, `none` for no assignee/suggestion, `my_teams` for `#team-name` for teams you belong to.
```
Comment (barkbarkimashark) [inline src/docs/product/sentry-basics/search/searchable-properties/issues.mdx:35]: nit: maybe 'Returns issues that are assigned to or suggest to be assigned to defined user(s) or teams(s). '

**Review by barkbarkimashark — APPROVED** [review]

**Pull request merged**